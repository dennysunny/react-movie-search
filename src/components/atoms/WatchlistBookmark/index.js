import React, { Component } from 'react';

import { propTypes } from './propTypes';
import {
  removeMovieFromWatchlist,
  addMovieToWatchlist,
  FirebaseAuthContext,
} from '../../../firebase/FirebaseAuthProvider';

class WatchlistBookmark extends Component {
  static contextType = FirebaseAuthContext;

  static propTypes = propTypes;

  toggleMovie = () => {
    const { movie } = this.props;
    const { user } = this.context;

    (user.watchlist && user.watchlist[movie.id])
      ? removeMovieFromWatchlist(movie.id)
      : addMovieToWatchlist(movie);
  }

  render() {
    const { movie } = this.props;
    const { user } = this.context;
    const watchlist = user && user.watchlist && user.watchlist[movie.id];
    return (
      <div
        className="movie-add-watchlist__wrapper"
        onClick={() => this.toggleMovie()}
        onKeyDown={() => this.toggleMovie()}
        role="button"
        tabIndex="-1"
      >
        {user.email
          && (
            <span className={
              ['movie-add-watchlist__icon fa',
                (watchlist !== undefined)
                  ? 'fa-bookmark'
                  : 'fa-bookmark-o',
              ].join(' ')
            }
            />
          )
        }
      </div>
    );
  }
}

export default WatchlistBookmark;
