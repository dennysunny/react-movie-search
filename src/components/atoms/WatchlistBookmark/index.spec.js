import React from 'react';
import { shallow } from 'enzyme';

import WatchlistBookmark from '.';

jest.mock('firebase/app', () => ({
  apps: [{}],
  auth: jest.fn(function () {
    return {
      FacebookAuthProvider: jest.fn(),
    }
  }),
  database: jest.fn(),
}));

describe('Atoms/WatchlistBookmark', () => {
  // beforeAll(() => {
  //   firebase.firebase.auth = jest.fn().mockReturnValue({
  //     currentUser: true,
  //     signOut: () => true,
  //   });
  // });

  it('should render as expected', () => {
    const wrapper = shallow(<WatchlistBookmark />);

    expect(wrapper).toMatchSnapshot();
  });
});
