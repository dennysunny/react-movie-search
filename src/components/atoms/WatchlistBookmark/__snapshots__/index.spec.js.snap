// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/WatchlistBookmark should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WatchlistBookmark
    movie={
      Object {
        "id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "movie-add-watchlist__wrapper",
      "onClick": [Function],
      "onKeyDown": [Function],
      "role": "button",
      "tabIndex": "-1",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "movie-add-watchlist__wrapper",
        "onClick": [Function],
        "onKeyDown": [Function],
        "role": "button",
        "tabIndex": "-1",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
